multidict/_multidict_base.py

```diff
@@ -73,59 +77,43 @@ def _viewbaseset_richcmp(
 
 
 def _viewbaseset_and(
-    view: set[object], other: object
-) -> Union[set[object], NotImplementedType]:
+    view: _ViewArg, other: object
+) -> Union[set[Sequence[object]], NotImplementedType]:
     if not isinstance(other, Iterable):
         return NotImplemented  # type: ignore[no-any-return]
-    if isinstance(view, Set):
-        view = set(iter(view))
-    if isinstance(other, Set):
-        other = set(iter(other))
-    if not isinstance(other, Set):
-        other = set(iter(other))
-    return view & other
+    lft = set(iter(view))
+    rgt = set(iter(other))
+    return lft & rgt
 
 
 def _viewbaseset_or(
-    view: set[object], other: object
-) -> Union[set[object], NotImplementedType]:
+    view: _ViewArg, other: object
+) -> Union[set[Sequence[object]], NotImplementedType]:
     if not isinstance(other, Iterable):
         return NotImplemented  # type: ignore[no-any-return]
-    if isinstance(view, Set):
-        view = set(iter(view))
-    if isinstance(other, Set):
-        other = set(iter(other))
-    if not isinstance(other, Set):
-        other = set(iter(other))
-    return view | other
+    lft = set(iter(view))
+    rgt = set(iter(other))
+    return lft | rgt
 
 
 def _viewbaseset_sub(
-    view: set[object], other: object
-) -> Union[set[object], NotImplementedType]:
+    view: _ViewArg, other: object
+) -> Union[set[Sequence[object]], NotImplementedType]:
     if not isinstance(other, Iterable):
         return NotImplemented  # type: ignore[no-any-return]
-    if isinstance(view, Set):
-        view = set(iter(view))
-    if isinstance(other, Set):
-        other = set(iter(other))
-    if not isinstance(other, Set):
-        other = set(iter(other))
-    return view - other
+    lft = set(iter(view))
+    rgt = set(iter(other))
+    return lft - rgt
 
 
 def _viewbaseset_xor(
-    view: set[object], other: object
-) -> Union[set[object], NotImplementedType]:
+    view: _ViewArg, other: object
+) -> Union[set[Sequence[object]], NotImplementedType]:
```

Comment:
I don't think you want to add Sequence here. If other is tuple[int], then that return type is not correct.

Reply:
I'm not sure why mypy is allowing it though...

Reply:
Nevermind, you're right as these particular methods only return items that are in the left side.

Reply:
Technically, Any is the best type. I don't want typing here; it doesn't make any value and doesn't help mypy to find the misusage

Reply:
The return type is probably less relevant, but the parameter types help to check for and catch mistakes such as #1045.